src/gen/scope.rs

```diff
@@ -0,0 +1,7 @@
+use koopa::ir;
+
+pub enum Scope<'a, 'b> {
+    Program,
+    Function(&'a ir::Function),
+    BasicBlock(&'a ir::Function, &'b ir::BasicBlock),
```

Comment:
```suggestion
    BasicBlock(&'b ir::Function, &'b ir::BasicBlock),
```

`Function` lifetime should be equal to `BasicBlock`